src/main/java/edu/cornell/kfs/module/purap/dataaccess/impl/CuB2BDaoImpl.java

```diff
@@ -23,64 +26,35 @@ public class CuB2BDaoImpl extends B2BDaoImpl {
-            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-
-            conn.setDoInput(true);
-            conn.setDoOutput(true);
-            conn.setRequestMethod("POST");
-			String charSet = "UTF-8";
-			if (request.contains("MIME_BOUNDARY_FOR_ATTACHMENTS")) {
-				// KFSPTS-794 : for attachments
-				conn.setRequestProperty("Content-type", "multipart/related; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
-				charSet = "ISO-8859-1";
-				LOG.info("content-type is multipart/related; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
-			} else {
-				conn.setRequestProperty("Content-type", "text/xml");
-
-				LOG.info("content-type is text/xml");
-			}
-
-            OutputStream out = conn.getOutputStream();
-            OutputStreamWriter outw = new OutputStreamWriter(out, charSet);
-            outw.write(request);
-            outw.flush();
-            outw.close();
-            out.flush();
-            out.close();
-
-            InputStream inp = conn.getInputStream();
-
-            StringBuffer response = new StringBuffer();
-            int i = inp.read();
-            while (i >= 0) {
-                if (i >= 0) {
-                    response.append((char) i);
-                }
-                i = inp.read();
-            }
-            return response.toString();
-        }
-        catch (MalformedURLException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (ProtocolException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (UnsupportedEncodingException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
-        catch (IOException e) {
-            LOG.error("postPunchOutSetupRequestMessage() Error posting setup", e);
-            throw new B2BConnectionException("Unable to connect to remote site for punchout.", e);
-        }
+    public String sendPunchOutRequest(final String request, final String punchoutUrl) {
+    	 LOG.debug("sendPunchOutRequest() started");
+
+         try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+             final HttpPost httpPost = new HttpPost(punchoutUrl);
+             if (request.contains("MIME_BOUNDARY_FOR_ATTACHMENTS")) {
+            	// KFSPTS-794 : for attachments
+                 httpPost.addHeader(HttpHeaders.ACCEPT, MediaType.MULTIPART_RELATED);
+                 httpPost.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.MULTIPART_RELATED + "; boundary=" + CUPurapConstants.MIME_BOUNDARY_FOR_ATTACHMENTS);
```

Comment:
After looking further at the Spring `MediaType` class, I noticed that it has a two-arg constructor that accepts a base `MediaType` and a parameter map. If you wanted to simplify this further, you could use that two-arg constructor and pass in the "boundary" setting using a singleton map.

Reply:
updated